server/server.go

```diff
@@ -362,6 +362,8 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		if status >= 400 {
 			DefaultErrorFunc(w, r, status)
 		}
+
+		w.WriteHeader(status)
```

Comment:
If the status code is >= 400, that means the middleware handlers have not written a response. Any other status code (0 according to the docs, but the way we've implemented it, could be anything else including 200) indicates that the middleware chain wrote to the response already.